packages/connections/src/stores/connections-store.ts

```diff
@@ -363,6 +365,39 @@ export function useConnections(
           });
         }
       },
+      async removeConnection(connectionInfo: ConnectionInfo) {
+        await connectionStorage.delete(connectionInfo);
+        dispatch({
+          type: 'set-connections',
+          connections: connections.filter(
+            (conn) => conn.id !== connectionInfo.id
+          ),
+        });
+        const nextActiveConnection = createNewConnectionInfo();
+        dispatch({
+          type: 'set-active-connection',
+          connectionId: nextActiveConnection.id,
+          connectionInfo: nextActiveConnection,
+        });
+      },
+      async duplicateConnection(connectionInfo: ConnectionInfo) {
+        const duplicate: ConnectionInfo = {
+          ...JSON.parse(JSON.stringify(connectionInfo)),
+          id: uuidv4(),
+        };
+        duplicate.favorite!.name += ' (copy)';
+
+        await saveConnectionInfo(duplicate);
+        dispatch({
+          type: 'set-connections',
+          connections: [...connections, duplicate],
+        });
+        dispatch({
+          type: 'set-active-connection',
```

Comment:
Oftentimes when two dispatches follow each other like here and above I find it might be better to create a new action and use just one action dispatch. Cleans up the code a bit and makes the action a bit more descriptive and devoid of following effects.

Reply:
Code is updated, I added `set-connections-and-select` action, let me know if you like this approach.

Reply:
Looks good!